diarymanagement-code-r273/tbaclient/tags/final.rc1/dbserver/src/TBA/Server/DBServer.java

```diff
@@ -0,0 +1,1118 @@
+        now.add(Calendar.DAY_OF_YEAR, -2);
+        soon.add(Calendar.DAY_OF_YEAR, -2);
+        executeUpdate("insert INTO entries(ID, DIARYID, START, END, USER, LOCKED, TITLE, BODY, DELETED) " + "values(6, 1, " + Long.toString(now.getTimeInMillis()) + ", " + Long.toString(soon.getTimeInMillis()) + ", 'dmcgra', 0, " + "'Group Meating', 'We are having a meeting about project Win', 0)");
+        now.add(Calendar.DAY_OF_YEAR, 4);
+        soon.add(Calendar.DAY_OF_YEAR, 4);
+        executeUpdate("insert INTO entries(ID, DIARYID, START, END, USER, LOCKED, TITLE, BODY, DELETED) " + "values(7, 0, " + Long.toString(now.getTimeInMillis()) + ", " + Long.toString(soon.getTimeInMillis()) + ", 'dmcgra', 0, " + "'Group Vegetabling', 'We are having a meeting about project TX-OneBillion', 0)");
+    }
+
+    /**
+     * This is the constructor for DBServer.
+     *<p>
+     *<p>
+     */
+    public void RealDatabase() {
+        // Drop Table from another
+
+
+        DropTables();
+        // Drop Triggers of the old System
+        DropTriggers();
+
+        // Creates All the tables
+        CreateTables();
+        // Creates The Triggers to Enforce Foreign Keys
+        CreateTriggers();
+    }
+
+    //generates the list of diaries for the task need
+    //returns the diary
+
+    public int SmartList(String usr, int type) {
+        //type 1 = permisions
+        //type 2 = diaries
+        String table, display;
+        try {
+            ResultSet rset = null;
+            if (type == 1) {
+                table = "Diary";
+                display = "dairy";
+                smartlistone.setString(1, usr);
+                rset = smartlistone.executeQuery();
+            } else if (type == 2) {
+                table = "Diaries";
+                display = "premission";
+                smartlisttwo.setString(1, usr);
+                rset = smartlisttwo.executeQuery();
+            } else {
+                return -1;
+            }
+
+            int list[] = null;
+            int i = 0;
+            System.out.println("Select a " + display + " owned by " + usr + " :");
+            // If A Result Is Returned
+            while (rset.next()) {
+                System.out.println("inside");
+                System.out.println("[" + i + "] : " + rset.getString(0));
+                System.out.println("inside");
+                list[i] = rset.getInt(1);
+                System.out.println("inside");
+                i++;
+            }
+
+            System.out.println("[-1] : None of the above (Cancle transation)");
+            System.out.println("$");
+            BufferedReader stdin = new BufferedReader(new InputStreamReader(System.in));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *stdin*. Currently, there are execution paths that do not contain closure statements. Either a) close *stdin* in a try-finally block or b) close the resource by declaring *stdin* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).